### ISSUE TITLE
Connect over SSH with key and username/password

### ISSUE THREAD

**Issue by soleares** [TablePlus/TablePlus#1221]
1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):
MySQL 5.X
2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):
228
3. The steps to reproduce this issue:
	- Create connect with "Over SSH"
	- Fill in username/password and select key

	Expected: Connects successfully
	Actual: Doesn't connect and shows the following error in the logs (using Debug version):
	```
	[2019/06/01 14:22:32.778932, 1] ssh_packet_userauth_failure:  Access denied for 'unknown'. Authentication that can continue: password
	[2019/06/01 14:22:32.778943, 2] ssh_packet_userauth_failure:  Access denied for 'unknown'. Authentication that can continue: password
	```

This works with Sequel Pro. It uses both the key and SSH username/password to connect to the bastion server.

Does TablePlus support this? It seems like it doesn't because in the UI when I select a private key it blanks out the password. I want it to use both like Sequel Plus. The logs seem to indicate that it connects with the key but then don't use the username/password to login - because my username is not 'unknown'.

Is there a way to support this? I just bought the product and would love to use it.

**Comment by huyphams** [comment]
TablePlus does respect the username input when using the SSH private key. Could you double check the ~/.ssh/config and make sure it doesn't override the input?

**Comment by soleares** [comment]
I confirmed the same behavior using a working ~/.ssh/config and leaving username/password blank and key unset in the UI as the SSH docs/issues specify.

I'm pretty sure the error message is just a fallback message after it failed authentication. I think the issue is that TablePlus doesn't send the SSH password if the SSH key is specified (IdentityFile in ~/.ssh/config).

The server is configured to require publickey **and** password authentication (common for 2FA but not 2FA in this case).
```
AuthenticationMethods "publickey,password" "publickey,keyboard-interactive"
```
If an SSH key is specified does TablePlus send the SSH password as well? I assume not because it's grayed out in the UI. Sequel plus does so it supports this configuration.

**Comment by mgoggin** [comment]
I can confirm the same issue. When Use SSH Key is checked, the password is not sent. At work, we started using TablePlus to connect to our read replica in production for debugging and not being able to save the password is an annoyance.

Any chance that might be supported in the future?

**Comment by mgoggin** [comment]
@huyphams I have a team of 10 using TablePlus and being impacted by this. What kind of feedback can we give you to assist with this issue?

**Comment by swznd** [comment]
if you got error message `Access Denied for 'unknown'`, make sure username doesn't contain space

@huyphams I think TablePlus should trim space

**Comment by otakuryo** [comment]
If nothing work properly
Create tunnel SSH to port 
 `ssh -L -N 8125:127.0.0.1:5432 USER-SSH@IP-SERVER -p PORT -i ~/.ssh/ssh_key` 

and connect to local port.

Note: If connect to docker sql server, attention to docker ip  127.17 
 `ssh -L -N 8125:127.17.0.1:PORT-DOCKER-SQL-INSTANCE USER-SSH@IP-SERVER -p PORT -i ~/.ssh/ssh_key` 

> (Ex. 0.0.0.0:22371->5432/tcp >> 0.0.0.0: PORT-DOCKER->MACHINE-PORT/tcp)
<img width="612" alt="Captura de pantalla 2020-08-09 a las 5 57 00" src="https://user-images.githubusercontent.com/12984409/89724580-42e1b000-da05-11ea-9e8d-7797f36cbe6d.png">

**Comment by soleares** [comment]
@otakuryo Yes, this can be worked around by tunneling but I filed this issue to try to get this resolved so that the extra step of tunneling isn’t necessary like in Sequel Pro.

**Comment by otakuryo** [comment]
> @otakuryo Yes, this can be worked around by tunneling but I filed this issue to try to get this resolved so that the extra step of tunneling isn’t necessary like in Sequel Pro.

Yes, it does't repair the problem, it is temporary patch, personally, this patch saved my day

**Comment by gavi89** [comment]
how is it possible that we are in 2022 and the bug has not yet been solved? There's news about that ?

**Comment by jpmc** [comment]
Also wanted to report in I have the same issue still. My private key has a password, which it takes, but then instead of then proceeding to the next password it just errors out.

Additional context: Our SSH uses both key+password, my key just so happens to also have a password on top of that.

Expected behavior: Enter the password for the key, then a second password prompt for the SSH connection itself should appear.

**Comment by nousename** [comment]
4 years later still seems to be a problem
